tests/test_computed_fields.py

```diff
@@ -0,0 +1,482 @@
+    assert m.model_dump() == {'x': 2, 'double': 4}
+
+
+def test_private_computed_field():
+    class MyModel(BaseModel):
+        x: int
+
+        @computed_field
+        def _double(self) -> int:
+            return self.x * 2
+
+    m = MyModel(x=2)
+    assert repr(m) == 'MyModel(x=2, _double=4)'
+    assert m.__private_attributes__ == {}
+    assert m._double == 4
+    assert m.model_dump() == {'x': 2, '_double': 4}
+
+
+@pytest.mark.skipif(sys.version_info < (3, 9), reason='fails before 3.9 - Do we want to fix this???')
+def test_classmethod():
+    class MyModel(BaseModel):
+        x: int
+        y: ClassVar[int] = 4
+
+        @computed_field
+        @classmethod
+        @property
+        def two_y(cls) -> int:
+            return cls.y * 2
+
+    m = MyModel(x=1)
+    assert m.two_y == 8
+    assert m.model_dump() == {'x': 1, 'two_y': 8}
+
+
+def test_frozen():
+    class Square(BaseModel, frozen=True):
+        side: float
+
+        @computed_field
+        @property
+        def area(self) -> float:
+            return self.side**2
+
+        @area.setter
+        def area(self, new_area: int):
+            self.side = new_area**0.5
+
+    m = Square(side=4)
+    assert m.area == 16.0
+    assert m.model_dump() == {'side': 4.0, 'area': 16.0}
+
+    with pytest.raises(TypeError, match='"Square" is frozen and does not support item assignment'):
+        m.area = 4
+
+
+def test_validate_assignment():
+    class Square(BaseModel, validate_assignment=True):
+        side: float
+
+        @field_validator('side')
+        def small_side(cls, s):
+            if s < 2:
+                raise ValueError('must be >=2')
+            return float(round(s))
+
+        @computed_field
+        @property
+        def area(self) -> float:
+            return self.side**2
+
+        @area.setter
+        def area(self, new_area: int):
+            self.side = new_area**0.5
+
+    with pytest.raises(ValidationError, match=r'side\s+Value error, must be >=2'):
+        Square(side=1)
+
+    m = Square(side=4.0)
+    assert m.area == 16.0
+    assert m.model_dump() == {'side': 4.0, 'area': 16.0}
+    m.area = 10.0
+    assert m.side == 3.0
+
+    with pytest.raises(ValidationError, match=r'side\s+Value error, must be >=2'):
+        m.area = 3
+
+
+def test_abstractmethod():
+    class AbstractSquare(BaseModel):
+        side: float
+
+        @computed_field
+        @property
+        @abstractmethod
+        def area(self) -> float:
+            raise NotImplementedError()
+
+    class Square(AbstractSquare):
+        @computed_field
+        @property
+        def area(self) -> float:
+            return self.side + 1
+
+    m = Square(side=4.0)
+    assert m.model_dump() == {'side': 4.0, 'area': 5.0}
+
+
+@pytest.mark.xfail(reason='missing abstractmethods are not raising an error')
+def test_abstractmethod_missing():
+    class AbstractSquare(BaseModel):
+        side: float
+
+        @computed_field
```

Comment:
This can be fixed by making the `computed_field` decorator check if `__isabstractmethod__` is True on the method it is being applied to. If it is True, make sure the respective computed field attribute has the attribute `__isabstractmethod__` set to True.

Reply:
Implement @adriangb suggestion, and check if it is still not working before going down this router.

Reply:
As I've said a few times in this thread, dynamically setting these methods doesn't work, at least for `__set_name__`.

Reply:
I don't know where you sadi that but yes you are right, to my surprise that doesn't work. Not even overriding `__getattr__` works for those methods. It does work for `__isabstractmethod__` and I just pushed that.